### PR TITLE
[fx] PassResult returned from passes

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81269
* __->__ #80532
* #80531

Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

### CODE REVIEW

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

File: torch/fx/passes/infra/pass_manager.py [hunk 1]
```diff
@@ -133,7 +155,7 @@ class PassManager:
             after each pass
     """
 
-    passes: List[Callable[[nn.Module], nn.Module]] = []
+    passes: List[Callable[[nn.Module], PassResult]] = []
```
Comment (SherlockNoMad) [inline torch/fx/passes/infra/pass_manager.py:158]: You mentioned some passes works with nn.Module, could you please share a code pointer for such case. 

I am thinking if we should constrain the PassBase to only work with GraphModule.

**Review by SherlockNoMad — APPROVED** [review]
LGTM, we can resolve "whether Pass should only work with GraphModule" in the latter PRs.

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

**Comment by angelayi** [comment]
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

**Pull request closed without merging**